Src/ILGPU/Runtime/ExchangeBuffer.cs

```diff
@@ -165,35 +117,40 @@ protected override void Dispose(bool disposing)
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
         private readonly ViewPointerWrapper cpuMemory;
 
+        /// <summary>
+        /// Property for accessing cpuMemory.
+        /// </summary>
+        protected ViewPointerWrapper CPUMemory => cpuMemory;
+
         /// <summary>
         /// A cached version of the CPU memory pointer.
         /// </summary>
         [DebuggerBrowsable(DebuggerBrowsableState.Never)]
         private readonly void* cpuMemoryPointer;
 
         /// <summary>
-        /// Initializes this memory buffer.
+        /// Property for accessing cpuMemoryPointer.
         /// </summary>
-        /// <param name="buffer">The underlying memory buffer.</param>
-        /// <param name="mode">The current buffer allocation mode.</param>
-        internal ExchangeBuffer(MemoryBuffer<T, Index1> buffer, ExchangeBufferMode mode)
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+        [CLSCompliant(false)]
+        protected void* CPUMemoryPointer => cpuMemoryPointer;
+
+        /// <summary>
+        /// Constructs the base class for all exchange buffer implementations.
+        /// </summary>
+        /// <param name="buffer">The memory buffer to use.</param>
+        /// <param name="mode">The exchange buffer mode to use.</param>
+        internal ExchangeBufferBase(MemoryBuffer<T, TIndex> buffer,
+            ExchangeBufferMode mode)
             : base(buffer.Accelerator, buffer.Extent.Size)
         {
-            // Allocate CPU memory
-            cpuMemory = Accelerator is CudaAccelerator &&
-                mode == ExchangeBufferMode.PreferPagedLockedMemory
-                ? CudaViewSource.Create(buffer.LengthInBytes)
-                : (ViewPointerWrapper)UnmanagedMemoryViewSource.Create(
-                    buffer.LengthInBytes);
+            cpuMemory = buffer.Accelerator is CudaAccelerator &&
+                    mode == ExchangeBufferMode.PreferPagedLockedMemory
```

Comment:
nit: Feels like the wrong indentation level; 4 spaces instead?

Reply:
I left this how it was in the original ExchangeBuffer.cs if I remember correctly. Should it be changed back?

Reply:
I simply changed this because it looked cleaner to me and I didn't understand the point of returning a reference here.

Reply:
I guess the indentation of `mode == ...` should be changed in this case because it has a relative indentation of 8 chars in comparison to the line `cpuMemory = ...`

Reply:
`mode == ... ` should be 4 spaces ahead of `cpuMemory =...`?